lib/routes/account.js

```diff
@@ -1505,7 +1079,8 @@ module.exports = (log, db, mailer, Password, config, customs, checkPassword, pus
             function (emailRecord) {
               uid = emailRecord.uid
 
-              return checkPassword(emailRecord, authPW, request.app.clientAddress)
+              const password = new Password(authPW, emailRecord.authSalt, emailRecord.verifierVersion)
+              return signinUtils.checkPassword(emailRecord, password, request.app.clientAddress)
```

Comment:
This was a side-effect of pulling `new password` out of the `checkPassword` helper, we have to do it locally before each invocation.